src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.Targets.cs

```diff
@@ -0,0 +1,193 @@
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics(
+                // (8,10): error CS8793: A module initializer must be an ordinary method
+                //         [ModuleInitializer]
+                Diagnostic(ErrorCode.ERR_ModuleInitializerMethodMustBeOrdinary, "ModuleInitializer").WithLocation(8, 10),
+                // (10,10): error CS8793: A module initializer must be an ordinary method
+                //         [ModuleInitializer]
+                Diagnostic(ErrorCode.ERR_ModuleInitializerMethodMustBeOrdinary, "ModuleInitializer").WithLocation(10, 10)
+                );
+        }
+
+        [Fact]
+        public void IgnoredOnPropertyAccessors()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+class C
+{
+    public int P 
+    {
+        [ModuleInitializer]
+        get;
+        [ModuleInitializer]
+        set;
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics(
+                // (8,10): error CS8793: A module initializer must be an ordinary method
+                //         [ModuleInitializer]
+                Diagnostic(ErrorCode.ERR_ModuleInitializerMethodMustBeOrdinary, "ModuleInitializer").WithLocation(8, 10),
+                // (10,10): error CS8793: A module initializer must be an ordinary method
+                //         [ModuleInitializer]
+                Diagnostic(ErrorCode.ERR_ModuleInitializerMethodMustBeOrdinary, "ModuleInitializer").WithLocation(10, 10)
+                );
+        }
+
+        [Fact]
+        public void IgnoredOnIndexerAccessors()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+class C
+{
+    public int this[int p]
+    {
+        [ModuleInitializer]
+        get => p;
+        [ModuleInitializer]
+        set { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics(
+                // (8,10): error CS8793: A module initializer must be an ordinary method
+                //         [ModuleInitializer]
+                Diagnostic(ErrorCode.ERR_ModuleInitializerMethodMustBeOrdinary, "ModuleInitializer").WithLocation(8, 10),
+                // (10,10): error CS8793: A module initializer must be an ordinary method
+                //         [ModuleInitializer]
+                Diagnostic(ErrorCode.ERR_ModuleInitializerMethodMustBeOrdinary, "ModuleInitializer").WithLocation(10, 10)
+                );
+        }
+    }
```

Comment:
>} [](start = 4, length = 1)

It doesn't look like there is a test for constructors.

Reply:
The tests are in `.Ignored.cs`: `IgnoredOnInstanceConstructor` and `IgnoredOnStaticConstructor`. I split the tests between .Targets and .Ignored according to the way AttributeTargets sees them.